### PR TITLE
ENH: tool for migrating TwinCAT repositories to latest standards

### PR DESCRIPTION
**Pull request by klauer**
Lazily borrows "fixes" from the Python code migration tools

Updater:
* Moves to GitHub Actions
* Updates LICENSE
* Updates pre-commit-config
* Updates README (if not yet existing)
* Updates `.gitignore` and `.gitattributes`
* Adds issue/PR templates (these are skewed toward PLC projects and not libraries; may need adjusting)

### CODE REVIEW

**Comment by klauer** [comment]
(I just requested 100 reviews from the reviewers, what's one more?)

**Comment by tangkong** [comment]
I saw this one and skipped it intentionally 👀

File: update_twincat_repository.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    )
+
+    if remove_docs_tokens.files:
+        fixes.append(remove_docs_tokens)
+
+    fixes.extend(
+        [
+            helpers.PrecommitAutoupdate(name="run_precommit", repo=repo),
+            # NOTE: don't run pre-commit; users will rebel
+            # RunPrecommit(name="run_precommit", repo=repo),
```
Comment (tangkong) [inline update_twincat_repository.py:169]: 😢

File: migrate_single_twincat.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+git remote -v
+git push -u -v --force "$USER" "${BRANCH_NAME}"
+gh pr create \
+  --title "CI: Migrate to GitHub Actions" \
+  --body-file "$SCRIPT_DIR/templates/twincat/migration_pr.md" \
+  --base master \
+  --repo "$REPO" \
+  --reviewer zllentz \
+  --reviewer tangkong
```
Comment (tangkong) [inline migrate_single_twincat.sh:67]: I was wondering how this part was automated.  Neato

**Review by tangkong — APPROVED** [review]
I mean I just saw this work so I don't think I have any grounds for rejection.  

Nice work 👍

**Pull request merged**